apps/src/templates/sectionsRefresh/QuickAssignTable.jsx

```diff
@@ -15,10 +15,42 @@ export default function QuickAssignTable({
   courseOfferings,
   setSelectedCourseOffering,
   updateCourse,
-  sectionCourse
+  sectionCourse,
+  isNewSection
 }) {
   // Key is type of curriculum e.g. 'Course' or 'Module', which is the singular
   // version of the title we want for the column
+
+  useEffect(() => {
+    if (!isNewSection) {
+      // combines all the data into one object
+      const startingData = {
+        ...courseOfferings[marketingAudience][curriculumTypes.course],
+        ...courseOfferings[marketingAudience][curriculumTypes.standalone_unit],
+        ...courseOfferings[marketingAudience][curriculumTypes.module]
+      };
+
+      const headers = Object.keys(startingData);
+      console.log(startingData);
+
+      // iterates over all courses to find the course assigned
+      headers.map(header => {
+        const courseDataByHeaderValues = Object.values(startingData[header]);
+        courseDataByHeaderValues.map(course =>
+          sectionCourse?.courseOfferingId === course.id
+            ? setSelectedCourseOffering(course)
+            : null
+        );
+      });
+    }
+  }, [
+    courseOfferings,
+    marketingAudience,
+    sectionCourse?.courseOfferingId,
+    setSelectedCourseOffering,
+    isNewSection
```

Comment:
>  since `setSelectedCourseOffering` is a function, it shouldn't need to be in the dependencies

just saw some code that used a function in the dependencies, so I might be wrong about this! i'd be interested to see what you end up with, since i know you were looking into useEffect more

Reply:
Thanks!  Yes, I was getting a linting error when I tried committing without the function in the dependencies.  Bethany said in passing that she didn't love our linter for this reason, but I'd love to learn more about her thoughts on that so I can understand the dependencies more.  It is weird because it seems to work without dependancies listed.  I probably need to do a bit more self-study on this too, to understand the why behind it all.